dnf/base.py

```diff
@@ -2161,6 +2162,7 @@ def _prov_key_data(msg):
                     raise dnf.exceptions.Error(_prov_key_data(msg))
                 logger.info(_('Key imported successfully'))
                 key_installed = True
+                self._repo_set_imported_gpg_keys.add(repo.id)
```

Comment:
This code line will not be reached, if all retrieved keys are already installed.
I think that:
If all retrieved keys were already installed (misc.keyInstalled(self._ts, info.rpm_id, info.timestamp) >= 0 is True for all, look at some lines above) then code `self._repo_set_imported_gpg_keys.add(repo.id)` should be called too.